.github/release.yml

```
@@ -0,0 +1,9 @@
+---
+changelog:
+  categories:
+    - title: Backwards-incompatible protocol changes
+      labels:
+        - protocol-incompat
+    - title: Protocol changes
+      labels:
+        - protocol
```

I think the current release notes for Arica are too thin. We should display all changes and not just big things to the protocol. But maybe we should add two other labels:
```suggestion
        - protocol
    - title: Document changes
        - document
    - title: Other changes
        - "*"
```

---

_Two_ other labels? You mean two headings (categories in this config file), one of which using a new label, right?

Listing all commits seems pointless since that's what the commit history (that we already link to) is for, but a heading for documentation updates would be okay. Would that label cover documentation changes except protocol changes (which implicitly affect the docs), or what's the definition?

---

My intention with the additions above was to create two new heading (maybe I misread).

At least we should note changes to the documentation e.g. in Arica we had several additions that changed the index pages. Those I think should be listed.

When it comes to the other changes I don't understand listing them being pointless. In other projects I have found somewhere listing of changes they have done e.g [Argo Workflow Changelog](https://github.com/argoproj/argo-workflows/blob/master/CHANGELOG.md), [Firefox](https://www.mozilla.org/en-US/firefox/103.0/releasenotes/)

---

Yes, but we already link to the complete list of changes. If we single out protocol changes and doc updates what remains is basically script changes, tox.ini updates, GitHub Actions workflows etc and including them in the release notes is IMHO mostly a distraction. Those details are just a click away for those who want it.

---

At least add the documentation changes. I'll bend if no one supports me but I think its easier to read a list without all the dates and buttons (one-click away) also finding the smaller changes makes you flip forth and back as the "one-click away" list doesn't have any tags..

---

Suggestion: Put all "Others" into a collapsible field.

<details>
<summary>Other changes</summary>

- Bugfix 1
- Bugfix 2

</details>

---

How would that work in practice? If we want the generation to be automatic there's only this YAML file to play with. Or are you suggesting that we hand-edit the generated release notes to wrap the Others section in details/summary elements? (Not ruling out manual edits, just trying to understand.)

---

Thought that this document was a lot less strict. Since it does not allow to do what I suggested, I rescind my suggestion.

---

I've updated the [Arica edition release notes](https://github.com/eiffel-community/eiffel/releases/tag/edition-arica) with an "Other changes" section. I think it looks okay. The section isn't too long so the manual labor required to get a collapsible "Other changes" might not be worth it. If you're happy with how it looks I'll update the PR.

---

@m-linner-ericsson, @t-persson, @e-backmark-ericsson, would you care to have a look at the updated release notes example (linked in the prior comment)?

---

lgtm

---

👍

---

Okay, done.